### PR TITLE
Support aspect-ratio CSS for layout=responsive

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #30291

This pull request implements two features:

1. When `aspect-ratio` is supported, the `layout=responsive` is implemented via this property and the private `sizer` element is not added.
2. The optimized AMP page can supply both `sizer` and `aspect-ratio` and AMP Runtime will hide the `sizer` if the browser supports the `aspect-ratio` CSS.

Additionally added few storybooks for the responsive layouts, with and without aspect-ratio support.

### CODE REVIEW

File: css/ampshared.css [hunk 1]
```diff
@@ -155,6 +155,16 @@ i-amphtml-sizer {
   display: block !important;
 }
 
+/*
+ * Disable `i-amphtml-sizer` when aspect ratio is supported by both the
+ * borwser and the optimizer.
+ */
+@supports (aspect-ratio: 1 / 1) {
```
Comment (kristoferbaxter) [inline css/ampshared.css:162]: Have we validated this is completely ignored in unsupported browsers? Am not familiar with how older engines treat this at-rule.

File: test/unit/test-layout.js [hunk 2]
```diff
@@ -613,3 +615,91 @@ describe('Layout', () => {
...
+  });
+
+  afterEach(() => {
+    resetShouldUseAspectRatioCssForTesting();
+  });
+
+  describe('aspect-ratio not supported', function () {
+    before(function () {
+      if (CSS.supports('aspect-ratio: 1/1')) {
+        this.skipTest();
```
Comment (kristoferbaxter) [inline test/unit/test-layout.js:645]: Possible to ensure this test is run on something that doesn't support `aspect-ratio` as well?

I believe we support a method to ensure a test runs on ie11 for instance.

**Review by kristoferbaxter — APPROVED** [review]
Nice work, looks great.

A few small comments.

File: css/ampshared.css (hunk 1 above)
Comment (dvoytenko) [reply css/ampshared.css:162]: Yes. Works correctly everywhere, including IE.

File: test/unit/test-layout.js (hunk 2 above)
Comment (dvoytenko) [reply test/unit/test-layout.js:645]: Could you pls clarify? I'm not familiar with this mechanism. But FWIW, no browser that we use on Travis support aspect-ratio yet. I tested it locally on Chromium though.

**Comment by dvoytenko** [comment]
Just pushed the storybooks.

File: test/unit/test-layout.js (hunk 2 above)
Comment (kristoferbaxter) [reply test/unit/test-layout.js:645]: @rsimha How do we opt-in a specific test for running in older browsers?

File: css/ampshared.css [hunk 3]
```diff
@@ -155,6 +155,16 @@ i-amphtml-sizer {
...
+  i-amphtml-sizer.i-amphtml-disable-for-ar {
+    display: none !important;
```
Comment (jridgewell) [inline css/ampshared.css:164]: I think this is actually a regression. We took SSR rendering with correct size, and his the thing that gave it size. We now have to wait for v0 to init, define, and connect the CE for size to come back.

**Review by jridgewell — CHANGES_REQUESTED** [review]

File: test/unit/test-layout.js (hunk 2 above)
Comment (rsimha) [reply test/unit/test-layout.js:645]: IE is the only browser on which tests are opt-in. Here's how you can run a test on IE:

https://github.com/ampproject/amphtml/blob/a702b776a7807d446568b0e2309fcd2998c92319/test/integration/test-amp-img.js#L157-L162

File: css/ampshared.css (hunk 3 above)
Comment (dvoytenko) [reply css/ampshared.css:164]: The idea is that the optimizer will do two things:
1. Set `style="aspect-ratio: width / height"`.
2. Add `i-amphtml-disable-for-ar` class to the sizer.

Runtime will only disable `i-amphtml-disable-for-ar` when it supports `aspect-ratio`. Thus there shouldn't be any regressions. Optimizer will simply gives us two "correct" sizes and we don't need them both.

File: test/unit/test-layout.js (hunk 2 above)
Comment (dvoytenko) [reply test/unit/test-layout.js]: Thanks! And done!

File: css/ampshared.css (hunk 3 above)
Comment (jridgewell) [reply css/ampshared.css:164]: Ahh, ok.

**Review by jridgewell — APPROVED** [review]

**Pull request merged**